### PR TITLE
SSCS-8867: Notification – Reminders are scheduled to run at 12am

### PR DESCRIPTION
**Pull request by srochani**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java [hunk 1]
```diff
@@ -23,8 +33,8 @@
                 return Optional.of(event.getValue().getDateTime());
             }
         }
-
-        return Optional.ofNullable(caseData.getDwpResponseDate()).map(date -> ZonedDateTime.parse(date + "T00:00:00Z").toInstant().atZone(ZoneId.of(ZONE_ID)));
+        LocalTime time = LocalTime.MIN.plusSeconds(rand.nextInt(60 * 60 * 1000));
```
Comment (danhamilton) [inline src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java:36]: Is * 1000 (for ms?) here correct given you are using plus seconds?

File: src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java [hunk 2]
```diff
@@ -23,8 +33,8 @@
...
+        return Optional.ofNullable(caseData.getDwpResponseDate()).map(date -> ZonedDateTime.parse(date + "T" + time.toString() + "Z").toInstant().atZone(ZoneId.of(ZONE_ID)));
```
Comment (danhamilton) [inline src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java:37]: question: Intention here is to give a randomness to the time element DWP Response Date which will vary within what bounds?

Comment (srochani) [reply src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java:37]: Yes, at the moment I bounds the times 24 hours

File: src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java (hunk 1 above)
Comment (srochani) [reply src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java:36]: its just number, random will pick the value and bound to 24hours

Comment (danhamilton) [reply src/main/java/uk/gov/hmcts/reform/sscs/extractor/DwpResponseReceivedDateExtractor.java:36]: Should it not be 60 * 60 * 24 in that case?

60*60*24 = 86,400 = seconds in a day
60*60*1000 = 3,600,000.

For 60*60*1000 76% of the rands generated will be above 86,400 seconds in the day and therefore all schedule at 23:59

**Review by danhamilton — APPROVED** [review]

**Pull request merged**